src/posix/platform/radio_spinel.cpp

```diff
@@ -207,6 +205,22 @@ void RadioSpinel::Init(const char *aRadioFile, const char *aRadioConfig)
     SuccessOrDie(error);
 }
 
+otError RadioSpinel::ResetRCP(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    SuccessOrExit(error = SendReset());
+    SuccessOrExit(error = WaitResponse());
+    VerifyOrExit(mIsReady, error = OT_ERROR_FAILED);
+
+    SuccessOrExit(error = CheckSpinelVersion());
```

Comment:
How about keeping these checks in `RadioSpinel::Init()`, since I guess we don't need to check the version/capabilities from destructor call? We probably just want to reset the RCP in that case.

Thoughts?

Reply:
Updated accordingly